scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -248,14 +248,15 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
    * @param hotKeyThreshold key with `hotKeyThreshold` values will be considered hot. Some runners
    *                        have inefficient GroupByKey implementation for groups with more than 10K
    *                        values. Thus it is recommended to set `hotKeyThreshold` to below 10K,
-   *                        keep upper estimation error in mind.
+   *                        keep upper estimation error in mind. If you sample input via
+   *                        `sampleFraction` make sure to adjust `hotKeyThreshold` accordingly.
```

Comment:
maybe a bit more specific, `accordingly by multiplying the raw frequency estimate with sampleFraction`